### PR TITLE
#167728034 Complete endpoint for user can delete comment posted within comment thread

### PR DESCRIPTION
**Pull request by richienabuk**
**What does this PR do?**

  Create endpoint for user to delete comment /api/v1/comment/trips/uuid

  **Description of Task to be completed?**

  Create a DELEETE route to api/v1/comment/trips/:uuid
  Create a method in comment controller to handle the logic to update a comment

  **How should this be manually tested?**

  - Clone repository
  - Run `cp .env-example .env` and edit with approapriate config
  - `npm install`
  -` npm run db:migrate`
  -` npm run db:seed:all`
  - Login with seeded user and visit endpoint `/api/v1/trips`

  **Any background context you want to provide?**

- The feature requires that a user is able to delete a comment but the comment is not removed from the database.
- To achieve this, soft-delete approach was taken.

  **What are the relevant stories?**

  [#167728034](https://www.pivotaltracker.com/n/projects/2381984/stories/167728034)

  **Screenshots (if appropriate)**

 
delete comment

### CODE REVIEW

**Review by Abobos — APPROVED** [review]

**Review by Wokoro — APPROVED** [review]
Greate work man. LGTM.

**Review by dieudonneAwa — APPROVED** [review]

**Review by justinefe — APPROVED** [review]

**Review by justinefe — APPROVED** [review]

File: src/repositories/CommentRepository.js [hunk 1]
```diff
@@ -28,6 +28,56 @@ class CommentRepository {
...
+  }
+
+  /**
+   * @description deletes a users comment
+   *
+   * @param {object} condition contains the reference to the comment to delete
+   *
+   * @returns {boolean} returns true when delete successful
+   */
+  async delete(condition) {
```
Comment (chukwuemekachm) [inline src/repositories/CommentRepository.js:74]: Are we going with hard or soft delete for comments?

**Review by chukwuemekachm — COMMENTED** [review]
Just a quick question.

File: src/repositories/CommentRepository.js (hunk 1 above)
Comment (richienabuk) [reply src/repositories/CommentRepository.js:74]: The PT story required that the comment is removed from the frontend but not from the database.

Comment (chukwuemekachm) [reply src/repositories/CommentRepository.js:74]: Okay.

**Review by chukwuemekachm — APPROVED** [review]
LGTM.

**Pull request merged**